File: mpp-proiect-csharp-culbec/Model/Person.cs
using Model;

namespace Model;

public class Person<TId> : Entity<TId>
{
    public virtual string? FirstName { get; set; }

    public virtual string? LastName { get; set; }

    internal Person()
    {

    }

    public new class Builder
    {
        private Person<TId> _member = new Person<TId>();

        public Builder Reset()
        {
            _member = new Person<TId>();
            return this;
        }

        public Builder SetId(TId id)
        {
            _member.Id = id;
            return this;
        }

        public Builder SetFirstName(string firstName)
        {
            _member.FirstName = firstName;
            return this;
        }

        public Builder SetLastName(string lastName)
        {
            _member.LastName = lastName;
            return this;
        }

        public Person<TId> Build()
        {
            return _member;
        }
    }

    public new static Builder NewBuilder()
    {
        return new Builder();
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        return base.Equals(obj) && obj.GetType() == GetType() && Equals(FirstName, ((Person<TId>)obj).FirstName) &&
               Equals(LastName, ((Person<TId>)obj).LastName);
    }

    public override int GetHashCode()
    {
        return base.GetHashCode() + EqualityComparer<string>.Default.GetHashCode(FirstName!) +
               EqualityComparer<string>.Default.GetHashCode(LastName!);
    }

    public override string ToString()
    {
        return $"{base.ToString()}, {nameof(FirstName)}: {FirstName}, {nameof(LastName)}: {LastName}";
    }
}

File: mpp-proiect-java-culbec/MotorcycleContest/Model/src/main/java/Model/Person.java
package Model;

import java.util.Objects;

public class Person<ID> extends Entity<ID> {
    private String firstName = null;
    private String lastName = null;

    Person() {
        super();
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public static class Builder<ID> {
        public Builder() {
        }

        private Person<ID> member = new Person<>();

        public Builder<ID> reset() {
            this.member = new Person<>();
            return this;
        }

        public Builder<ID> setId(ID id) {
            this.member.setId(id);
            return this;
        }

        public Builder<ID> setFirstName(String firstName) {
            this.member.firstName = firstName;
            return this;
        }

        public Builder<ID> setLastName(String lastName) {
            this.member.lastName = lastName;
            return this;
        }

        public Person<ID> build() {
            return member;
        }
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        Person person = (Person) o;
        return Objects.equals(firstName, person.firstName) && Objects.equals(lastName, person.lastName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), firstName, lastName);
    }

    @Override
    public String toString() {
        return "Model.Person{" +
                "firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                "} " + super.toString();
    }
}
